docs/source/configuration/authorization.mdx

```diff
@@ -98,199 +165,278 @@ So the `me` top level operation would not even be executed. When returning the r
 }
 ```
 
-If all the fields were removed, then the router would generate a query planner error indicating that the query is unauthorized.
+If _every_ requested field requires authentication and a request is unauthenticated, the router generates an error indicating that the query is unauthorized.
+
+### `@requiresScopes`
+
+The `@requiresScopes` directive marks fields and types as restricted based on required scopes.
+The directive should include a `scopes` argument that defines an array of the required scopes.
+
+```graphql
+@requiresScopes(scopes: ["scope1", "scope2", "scope3"])
+```
+
+The directive validates the required scopes by loading the claims object at the `apollo_authentication::JWT::claims` key in a request's context.
+The claims object's `scope` key's value should be a space separated string of scopes in the format defined by the [OAuth2 RFC for access token scopes](https://datatracker.ietf.org/doc/html/rfc6749#section-3.3).
 
-## `@requiresScopes`
+```
```

Comment:
I found this useful to understand the shape/types expected in claims/scopes. What's the right code type to mark it as?

Reply:
We can use js, rhai is close enough on the syntax